### PR TITLE
List orders paid with a particular payment mean on admin orders page …

### PR DESCRIPTION
**Pull request by ithiame**
…#1340

### CODE REVIEW

File: src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import { collections } from '$lib/server/database';
+import { paymentMethods } from '$lib/server/payment-methods.js';
 import { ORDER_PAGINATION_LIMIT } from '$lib/types/Order';
 
-export async function load({ url }) {
+export async function load({ url, locals }) {
 	const skip = url.searchParams.get('skip');
 	const orderNumber = url.searchParams.get('orderNumber');
 	const productAlias = url.searchParams.get('productAlias');
+	const paymentMethod = url.searchParams.get('paymentMethod');
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts:9]: Please use zod validation on those parameters

File: src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts [hunk 2]
```diff
@@ -1,10 +1,19 @@
 import { collections } from '$lib/server/database';
+import { paymentMethods } from '$lib/server/payment-methods.js';
 import { ORDER_PAGINATION_LIMIT } from '$lib/types/Order';
+import { z } from 'zod';
+
+export async function load({ url, locals }) {
+	const querySchema = z.object({
+		skip: z.string().optional(),
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts:8]: this is a number, use `z.number().int().min(0)` or something like that

File: src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
+		orderNumber: z.string().optional(),
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts:9]: Same comment

File: src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts [hunk 4]
```diff
@@ -1,10 +1,19 @@
...
+		productAlias: z.string().optional(),
+		paymentMethod: z.string().optional()
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts:11]: This should be `z.enum()`, with all the pssible payment methods

File: src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts [hunk 5]
```diff
@@ -15,6 +24,13 @@ export async function load({ url }) {
...
 			.sort({ createdAt: -1 })
 			.toArray();
+	} else if (paymentMethod) {
+		orders = await collections.orders
+			.find({
+				'payments.method': paymentMethod
+			})
+			.sort({ createdAt: -1 })
+			.toArray();
 	} else {
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts:34]: The parameters should be able to be combined, eg skip + payment method together. You can default `skip` to `0` if it's helpful

**Comment by Tirodem** [comment]
It works, but :

image
- currently we have no option to reset the filter : best would be to add an option "All payments" at the beginning of the list to make a no-filter

- I have the impression that filtered results aren't sliced in X pages
With payment filter :
image
With no filter :
image

File: src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts [hunk 6]
```diff
@@ -13,8 +12,8 @@ export async function load({ url, locals }) {
 		skip: z.number({ coerce: true }).int().min(0).optional().default(0),
 		orderNumber: z.number({ coerce: true }).int().min(0).optional(),
 		productAlias: z.string().optional(),
-		paymentMethod: z.enum(methods as [PaymentMethod, ...PaymentMethod[]]).optional(),
-		country: z.enum([...COUNTRY_ALPHA2S] as [CountryAlpha2, ...CountryAlpha2[]]).optional(),
+		paymentMethod: z.string().optional(),
+		country: z.string().optional(),
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts:16]: you can keep the enum, just add `""` as an option

File: src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts [hunk 7]
```diff
@@ -1,29 +1,50 @@
...
+
+export async function load({ url, locals }) {
+	const methods = paymentMethods({ role: locals.user?.roleId });
+
+	const querySchema = z.object({
+		skip: z.number({ coerce: true }).int().min(0).optional().default(0),
+		orderNumber: z.number({ coerce: true }).int().min(0).optional(),
+		productAlias: z.string().optional(),
+		paymentMethod: z.enum([...methods, ''] as [PaymentMethod, ...PaymentMethod[], '']).optional(),
+		country: z.enum([...COUNTRY_ALPHA2S, ''] as [CountryAlpha2, ...CountryAlpha2[], '']).optional(),
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/order/+page.server.ts:17]:
```suggestion
		paymentMethod: z.enum(['' as const, ...methods]).optional(),
		country: z.enum(['' as const, ...COUNTRY_ALPHA2S]).optional(),
```

**Pull request merged**